src/main/scala/cognite/spark/v1/RelationshipsRelation.scala

```diff
@@ -0,0 +1,156 @@
+import com.cognite.sdk.scala.v1._
+import fs2.Stream
+import org.apache.spark.sql.sources.{Filter, InsertableRelation}
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.{Row, SQLContext}
+
+class RelationshipsRelation(config: RelationConfig)(val sqlContext: SQLContext)
+    extends SdkV1Relation[Relationship, String](config, "relationships")
+    with InsertableRelation
+    with WritableRelation {
+  import CdpConnector._
+
+  override def schema: StructType = structType[Relationship]
+
+  override def toRow(a: Relationship): Row = asRow(a)
+
+  override def uniqueId(a: Relationship): String = a.externalId
+
+  override def getStreams(filters: Array[Filter])(
+      client: GenericClient[IO],
+      limit: Option[Int],
+      numPartitions: Int): Seq[Stream[IO, Relationship]] = {
+    val fieldNames =
+      Array(
+        "sourceExternalId",
+        "sourceType",
+        "subtype",
+        "targetExternalId",
+        "targetType",
+        "minStartTime",
+        "maxStartTime",
+        "minEndTime",
+        "maxEndTime",
+        "minActiveAtTime",
+        "maxActiveAtTime",
+        "minConfidence",
+        "maxConfidence",
+        "dataSetId",
+        "labels",
+        "minCreatedTime",
+        "maxCreatedTime",
+        "minLastUpdatedTime",
+        "maxLastUpdatedTime"
+      )
+    val pushdownFilterExpression = toPushdownFilterExpression(filters)
+    val shouldGetAllRows = shouldGetAll(pushdownFilterExpression, fieldNames)
+    val filtersAsMaps = pushdownToParameters(pushdownFilterExpression)
+
+    val relationshipsFilterSeq = if (filtersAsMaps.isEmpty || shouldGetAllRows) {
+      Seq(RelationshipsFilter())
+    } else {
+      filtersAsMaps.distinct.map(relationshipsFilterFromMap)
+    }
+
+    val streamsPerFilter = relationshipsFilterSeq
+      .map { f =>
+        client.relationships.filterPartitions(f, numPartitions, limit)
+      }
+
+    // Merge streams related to each partition to make sure duplicate values are read into
+    // the same RDD partition
+    streamsPerFilter.transpose
+      .map(s => s.reduce(_.merge(_)))
+  }
+
+  def relationshipsFilterFromMap(m: Map[String, String]): RelationshipsFilter =
+    RelationshipsFilter(
+      sourceExternalIds = m.get("sourceExternalIds").map(stringSeqFromWrappedArray),
+      sourceTypes = m.get("sourceTypes").map(stringSeqFromWrappedArray),
+      targetExternalIds = m.get("targetExternalIds").map(stringSeqFromWrappedArray),
+      targetTypes = m.get("targetTypes").map(stringSeqFromWrappedArray),
+      dataSetIds = m.get("dataSetId").map(idsFromWrappedArray(_).map(CogniteInternalId)),
+      startTime = timeRangeFromMinAndMax(m.get("minStartTime"), m.get("maxStartTime")),
+      endTime = timeRangeFromMinAndMax(m.get("minEndTime"), m.get("maxEndTime")),
+      activeAtTime = timeRangeFromMinAndMax(m.get("minActiveAtTime"), m.get("maxActiveAtTime")),
+      labels = getLabelsFilter(m.get("labels")),
```

Comment:
The same goes for this. How can I convert string "map('containsAll', array('label1', 'label2'))" into a compatible data type? Had some troubles here.

Reply:
Good question! The desired en state here is to populate the `RelationshipsFilter` from the filters that Spark provide: `filters: Array[Filter]`. These filters are actually Spark translating the filter part of a query such as `"SELECT * FROM _CDF.ASSETS WHERE ASSETID IS IN('10', '50') AND EXTERNALID LIKE '%WELL'`, and we probably won't be able to parse queries like this into a "containsAll" / "containsAny" type of filter.. probably only the "containsAny" type will work?

Reply:
I think I missed what I would receive here in the first place.
For example, if the user queries this (of course if it is a possible case here):
"SELECT * FROM _CDF.RELATIONSHIPS WHERE "my_target_label" in LABELS"
what would I receive in m.get("labels")?

Reply:
I think,
this would stand for containsAny
"SELECT * FROM _CDF.RELATIONSHIPS WHERE 'my_target_label' IN LABELS OR  'my_target_label2' IN LABELS" 
this would work for containsAll
"SELECT * FROM _CDF.RELATIONSHIPS WHERE 'my_target_label' IN LABELS AND  'my_target_label2' IN LABELS" 

but I need to understand how it works and parse it accordingly.

Reply:
So the code we have now doesn't allow for that, and honestly I don't think we need to..
`filtersAsMaps` is a `Seq[Map[String, String]]`, where each Map will be used to build a `RelationshipsFilter`. We decode `OR`s into multiple Maps here, so for example
```
LABELS = 'my_target_label' OR LABELS  = 'my_target_label2'
```
will be decoded into
```
Seq(
Map("label" -> "my_target_label"),
Map("label" -> "my_target_label2
```
these are then transformed into `RelationshipsFilter`s, but we can't implement OR from here, and so not containsAny.

The good thing is that these filters are only used to build the queries we run to CDF, and Spark actually filters that result again afterwards! So we can feel free to read too many rows, and could even skip the labels filter here entirely without getting the wrong result. It would only impact performance, and in the case or Relationships, probably not very much.

Reply:
I got the idea, thanks! I noticed that it was not possible for now. I only implemented for containsAll and changed the approach a bit in general.